esrally/mechanic/launcher.py

```diff
@@ -297,10 +364,12 @@ def __init__(self, cfg, metrics_store, races_root_dir, clock=time.Clock):
         self._clock = clock
         self.races_root_dir = races_root_dir
         self.keep_running = self.cfg.opts("mechanic", "keep.running")
+        self.daemon = self.cfg.opts("mechanic", "daemon", default_value=False, mandatory=False)
```

Comment:
That's definitely a good idea for testing in the WIP PR but I think we can fully switch to daemonized startup once this work is finished. While it is not mentioned in the [docs for Elasticsearch 1.7](https://www.elastic.co/guide/en/elasticsearch/reference/1.7/setup.html), daemonized startup and writing to a PID file is supported since then:

```
daniel@io:distributions/elasticsearch-1.7.6 $ ./bin/elasticsearch --help
Usage: ./bin/elasticsearch [-vdh] [-p pidfile] [-D prop] [-X prop]
Start elasticsearch.
    -d            daemonize (run in background)
    -p pidfile    write PID to <pidfile>
    -h
    --help        print command line options
    -v            print elasticsearch version, then exit
    -D prop       set JAVA system property
    -X prop       set non-standard JAVA system property
   --prop=val
   --prop val     set elasticsearch property (i.e. -Des.<prop>=<val>)
```

In other words: We can rely that this works consistently across all versions.

Reply:
My intent was actually to allow the original "inproc" codepath to still be used, if desired.  But I don't know if that will have any value.  What do you think?

Reply:
IMHO we don't need to keep the other approach as well.